source/wrangling.md

```diff
@@ -1152,1262 +960,890 @@ glue("census_popn", "{0:,.0f}".format(35151728))
+> see an input variable `skipna`, which by default is set to `skipna=True`. This means that
+> `pandas` will skip `NaN` values when computing statistics.
+
+### Calculating summary statistics on dataframes
 
-```{index} pandas.DataFrame; sort_values
+What if you want to calculate summary statistics on an entire dataframe? Well,
+it turns out that the functions in {numref}`tab:basic-summary-statistics`
+can be applied to a whole data frame!
+For example, we can ask for the number of rows that each column has using `count`.
+```{code-cell} ipython3
+region_lang.count()
+```
+Not surprisingly, they are all the same. We could also ask for the `mean`, but
+some of the columns in `region_lang` contain string data with words like `"Vancouver"`
+and `"Halifax"`---for these columns there is no way for `pandas` to compute the mean.
+So we provide the keyword `numeric_only=True` so that it only computes the mean of columns with numeric values. This
+is also needed if you want the `sum` or `std`.
+```{code-cell} ipython3
+region_lang.mean(numeric_only=True)
+```
+If we ask for the `min` or the `max`, `pandas` will give you the smallest or largest number
+for columns with numeric values. For columns with text, it will return the
+least repeated value for `min` and the most repeated value for `max`. Again,
+if you only want the minimum and maximum value for
+numeric columns, you can provide `numeric_only=True`.
+```{code-cell} ipython3
+region_lang.max()
+```
+```{code-cell} ipython3
+region_lang.min()
 ```
 
-As we saw in Chapter {ref}`intro`, 
-we can use the `.sort_values` function 
-to order the rows in the data frame by the values of one or more columns. 
-Here we pass the column name `most_at_home` to sort the data frame rows by the values in that column, in ascending order.
+Similarly, if there are only some columns that you would like to get summary statistics of,
+we can first use `loc[]` and then ask for the summary statistic. An example of this is illustrated in {numref}`fig:summarize-across`.
+Later, we will talk about how you can use a more general function, `apply`, to also accomplish this.
 
-```{code-cell} ipython3
-large_region_lang = (
-    tidy_lang[tidy_lang["most_at_home"] > 10000]
-    .loc[:, ["region", "language", "most_at_home"]]
-    .sort_values(by="most_at_home")
-)
+```{figure} img/summarize/summarize.003.jpeg
+:name: fig:summarize-across
+:figclass: caption-hack
 
-large_region_lang
+`loc[]` or `apply` is useful for efficiently calculating summary statistics on many columns at once. The darker, top row of each table represents the column headers.
 ```
 
+Lets say that we want to know
+the mean and standard deviation of all of the columns between `"mother_tongue"` and `"lang_known"`.
+We use `loc[]` to specify the columns and then `agg` to ask for both the `mean` and `std`.
 ```{code-cell} ipython3
-:tags: [remove-cell]
+region_lang.loc[:, "mother_tongue":"lang_known"].agg(["mean", "std"])
```

Comment:
```suggestion
region_lang["mother_tongue":"lang_known"].agg(["mean", "std"])
```

If we make the change I suggestion to this section above and only use loc for both filtering rows and selecting coluns at the same time